modules/support/src/evm.rs

```diff
@@ -0,0 +1,98 @@
+use codec::FullCodec;
+
+use pallet_evm::{ExitError, ExitSucceed, Precompile};
+use sp_runtime::traits::MaybeSerializeDeserialize;
+use sp_std::{convert::TryInto, fmt::Debug, marker::PhantomData, prelude::*, result};
+
+use orml_traits::MultiCurrency;
+
+pub trait AccountIdConversion<AccountId> {
+	fn from_eth_address(address: &[u8; 20]) -> AccountId;
+}
+
+/// The `MultiCurrency` impl precompile.
+///
+/// The 1st byte of `input` indicates action.
+/// - `0`: Query total issuance. In this case, the 2nd byte is `currency_id`.
```

Comment:
make a `#[repr(u8)] enum` instead using numbers

Reply:
I can create an `Action` enum and make match more readable, but as the `intput` from evm are numbers, in the documentation it might be better to keep the numbers so that it's more clear for solidity side.